### PR TITLE
Implement preprocessor

### PR DESCRIPTION
**Pull request by GiasemiSh**
Closes https://github.com/QMSTR/qmstr-all/issues/148

Test it with: https://github.com/QMSTR/qmstr-demo/pull/59

### CODE REVIEW

File: pkg/arbuilder/arbuilder.go [hunk 1]
```diff
@@ -112,8 +112,6 @@ func (a *ArBuilder) getResults() ([]*service.FileNode, error) {
 			inputFileNode := &service.FileNode{}
 			ext := filepath.Ext(inFile)
 			switch ext {
-			case ".o":
```
Comment (hemarkus) [inline pkg/arbuilder/arbuilder.go]: why is this deleted?

Comment (GiasemiSh) [reply pkg/arbuilder/arbuilder.go]: @hemarkus It's included in the default case.

**Comment by GiasemiSh** [comment]
We are not sure if we need to support the preprocessor.
I close the PR but we keep the branch for future reference.
